app/Http/Controllers/HomeController.php

```diff
@@ -23,6 +24,28 @@ public function __construct()
      */
     public function index()
     {
-        return view('home');
+        $user = auth()->user();
+        $following = $user->following()->get();
+        $followers = $user->followers()->get();
+        return view('home', compact('user', 'following', 'followers'));
+    }
+
+    public function edit()
+    {
+        $user = auth()->user();
+
+        return view('profile', compact('user'));
+    }
+
+    public function update(Request $request)
+    {
+        $user = auth()->user();
+
+        $user->update([
+            'name' => $request['name'],
+            'email' => $request['email'],
+            'password' => Hash::make($request['password']),
+        ]);
+        return redirect()->route('home');
```

Comment:
Please always add a space before a return
```suggestion
        ]);
        
        return redirect()->route('home');
```